Assets/Scripts/NavmeshHelper.cs

```diff
@@ -9,8 +9,9 @@
 /// provide a floor collision plane, to prevent the avatar falling through
 /// the world when it leaves the navmesh.
 /// </summary>
-public class NavmeshHelper : MonoBehaviour
+public class NavmeshHelper : MessageConsumer
```

Comment:
This should be able to inherit from MonoBehaviour and also implement your KeyframeMessageConsumer interface.

Reply:
Note: in the long run, this class probably doesn't need to be a MonoBehaviour. This was my initial implementation and it is messy.